File: src/main/java/systems/conduit/main/core/datastore/DatastoreBackend.java
package systems.conduit.main.core.datastore;

import lombok.AllArgsConstructor;
import lombok.Getter;
import systems.conduit.main.core.datastore.backend.MySQLBackend;

/**
 * The type of backend the datastore uses.
 *
 * @author Innectic
 * @since 12/30/2019
 */
@AllArgsConstructor
@Getter
public enum DatastoreBackend {
    MySQL(MySQLBackend.class);

    private Class<? extends Datastore> handler;
}

File: src/main/kotlin/systems/conduit/core/datastore/DatastoreBackend.kt
package systems.conduit.core.datastore

import systems.conduit.core.datastore.backend.MemoryBackend
import systems.conduit.core.datastore.backend.MySQLBackend

/**
 * The type of backend the datastore uses.
 *
 * @author Innectic
 * @since 12/30/2019
 */
enum class DatastoreBackend(val handler: Class<out DatastoreHandler>) {
    MySQL(MySQLBackend::class.java), Memory(MemoryBackend::class.java);
}
